src/ensembl/src/shared/eye-toggle-icon/EyeToggleIcon.tsx

```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+import classNames from 'classnames';
+
+import eyeOnIcon from './eye-on.svg';
+import eyeOffIcon from './eye-off.svg';
+
+import styles from './EyeToggleIcon.scss';
+
+type Props = {
+  iconStatus: string;
```

Comment:
Is it an 'on/off' string? Would it be better represented by a boolean (e.g. `disabled: boolean`)?

Reply:
The icon can have at least three statuses. The three I've come across are: on, off and disabled.

Reply:
Oh, a 'disabled' button is visually different from an 'off' button, huh? 🤔 Then the button should encounter for all three of these statuses.

Would an `enum` be better than a `string` then?

Reply:
Agree, it would be better than a `string`.

Reply:
It was implemented as a string in the track panel so I didn't change it. It's updated to boolean now.

Reply:
Well, if it's a boolean, then the prop should be renamed to `disabled` or `isDisabled` or something like that (`iconStatus` is not something that you would expect to be true or false). But as @imran-ebi pointed out, there are more than just 2 states of that button; so probably we should go with `enum`.

Reply:
Should we have `isDisabled` and `isActive` so that we can use it like:
<Button isDisabled />, <Button isActive />, <Button isDisabled isActive={false} />

Reply:
I'd still prefer it to be an `enum` as in the future we could have more than three states for this button.

Reply:
Cool, `enum` is fine. I think `disabled` needs to be separated out as we will not be calling the onClick function if it is.